ceph/ceph_admin/osd.py

```diff
@@ -127,3 +131,29 @@ def apply(self, config: Dict) -> None:
             sleep(interval)
 
         raise OSDServiceFailure("OSDs are not up and running in hosts")
+
+    # add osd on single device
+    def add(self, config: Dict) -> None:
```

Comment:
This becomes an anti design pattern... i.e. it breaks SOLID principle used in this module.

IMO, a separate `daemon` module extending from `orch` needs to be created. The daemon module could be similar to `orch` and be created using `orch` as a reference.

`daemon` module would have `mixins` such as `[add, redeploy, rm, start, stop, restart, reconfig]`. Like this module, a specific module for daemon would need to be created. In the newly created module, we could override the inherited AddMixin to use for non-standard options.

These modules have been created such that it has a one-to-one mapping with the cephadm commands.